### ISSUE TITLE
Questions on how to get th library working

### ISSUE THREAD

**Issue by MagicAndre1981** [dkornev/Xamarin.SignalR.WebSocketTransport#1]
With my Xamarin app I use SignalR (classic 2.2.3 not Core version) but it was only using Server Sent Events. I'm not satisfied with the performance and would like to get WebSocket working and look if performance is better. So I found your library, added it:

```
 await Connection.Start(new AutoTransport(httpClient,
                        new IClientTransport[]
                        {
                            new WebSocketTransport(httpClient),
                            new ServerSentEventsTransport(httpClient),
                        }));
```

but either this causes hangs or raises `System.Net.WebSockets.WebSocketException `

```
System.Net.WebSockets.WebSocketException (0x80004005): The remote party closed the WebSocket connection without completing the close handshake. ---> System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer. ---> System.Net.Sockets.SocketException: Connection reset by peer
  at System.Net.Sockets.Socket.EndReceive (System.IAsyncResult asyncResult) [0x00012] in <bcdc1df2b3724ab69797f3819a126346>:0 
  at System.Net.Sockets.NetworkStream.EndRead (System.IAsyncResult asyncResult) [0x00057] in <bcdc1df2b3724ab69797f3819a126346>:0 
   --- End of inner exception stack trace ---
```

As Signal Server I use a selfhost OWIN console application and if I set a breakpoint in `public void Configuration(IAppBuilder app)` in Startup class of the server, I see in server.Capabilities that websocket support is available.

I use Win10 v1709, VS2017 -15.6.6, Xamarin.Forms app (.net standard 2.0 as shared project, UWP with Min/Target 16299 to have .net standard 2.0 support, Android (Targetframework 8.1 Oreo, min Version set to API 23-Marshmallow), used port is whitelisted in Windows 10 Firewall which hosts the server.

Any ideas on how to get your library working?

**Comment by MagicAndre1981** [comment]
Some voodoo happened. Today I got it working. In the signalR log I see it it is used now

`SignalR.Transports.WebSocketTransport Verbose: 0 : Sending outgoing message. Connection id: GUID, transport: WebSocketTransport, message: foo`

And it looks like the exception happens during debug where some timeouts occur. 

The performance is better compared to SSE.

**Comment by MagicAndre1981** [comment]
a bit later I get a new exception:

```
System.MissingMethodException: Method not found: 'System.Security.Cryptography.X509Certificates.X509CertificateCollection Microsoft.AspNet.SignalR.Client.IConnection.get_Certificates()'.
   at Xamarin.SignalR.Transport.WebSocketWrapperRequest.PrepareRequest()
   at Xamarin.SignalR.Transport.WebSocketWrapperRequest..ctor(ClientWebSocket clientWebSocket, IConnection connection)
   at Xamarin.SignalR.Transport.WebSocketTransport.<PerformConnect>d__18.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at foo.<Connect>d__9.MoveNext()
```

**Comment by dkornev** [comment]
Hey @MagicAndre1981
Honestly after some time of using WebSocket transport for SignalR we decided to move back to LongPolling. WebSocket transport was not stable and reliable on Xamarin iOS and Android during background mode, OS (Android especially) was killing java socket connection, but mono socket was still in connected state and there was no way to determine if we are connected or not (only exception handling).

Regarding you last exception:
Please try to certificate validation callback somewhere once in you code
`ServicePointManager.ServerCertificateValidationCallback += (sender, certificate, chain, sslPolicyErrors) => true;`

**Comment by dkornev** [comment]
Yes. Be sure that it's called only once. It's better to put  it somewhere inside your app initialization method

**Comment by MagicAndre1981** [comment]
I've added it to the constructor of the App class of the shared project, but I still get this `System.MissingMethodException`.

**Comment by MagicAndre1981** [comment]
the issue seams to come from SignalR itself:

`System.MissingMethodException: 'Method not found: 'System.Security.Cryptography.X509Certificates.X509CertificateCollection Microsoft.AspNet.SignalR.Client.IConnection.get_Certificates()'.'


`

**Comment by dkornev** [comment]
Hm. We were using websocket transport for our apps and they worked, not ideally, but they did. Did see that exception before. I'll try to take a look at that this week. Will keep you posted

**Comment by MagicAndre1981** [comment]
I only get this with Websocket, not SSE. the exceptions happens in PrepareRequest when you access the `Certificates` property of `_connection`. If I exclude this call I get a HTTP_E_STATUS_SERVER_ERROR (exception 0x801901f4) in the server, but I haven't used the debugger on the OWIN/SignalR server exe to see what happens here.

**Comment by MagicAndre1981** [comment]
I gave it up on the old SignalR version and switched to [ASP .Net Core Version of SignalR](https://docs.microsoft.com/en-us/aspnet/core/signalr/dotnet-client?view=aspnetcore-2.1), here they already [include WebSocket support](https://github.com/aspnet/SignalR/blob/release/2.2/src/Microsoft.AspNetCore.Http.Connections/Internal/Transports/WebSocketsTransport.cs).
